docs/engineering.md

```diff
@@ -0,0 +1,595 @@
+  - [Shield-Studies-Addon-Utils (`studyUtils.jsm`)](#shield-studies-addon-utils-studyutilsjsm)
+  - [Legacy Add-ons](#legacy-add-ons)
+  - [Your Feature, with Variations](#your-feature-with-variations)
+- [<span id="shield-telemetry">Shield Telemetry Details</span>](#span-idshield-telemetryshield-telemetry-detailsspan)
+  - [Shield Study Life-Cycle Telemetry](#shield-study-life-cycle-telemetry)
+  - [Expected ping counts](#expected-ping-counts)
+  - [How Probes are Sent from `studyUtils.jsm`](#how-probes-are-sent-from-studyutilsjsm)
+  - [Send your own probes](#send-your-own-probes)
+  - [Defining Custom Study Endings](#defining-custom-study-endings)
+- [Viewing Sent Telemetry Probes](#viewing-sent-telemetry-probes)
+  - [client](#client)
+  - [Collector (example s.t.m.o query)](#collector-example-stmo-query)
+- [Engineering Side-by-Side (a/b) Feature Variations](#engineering-side-by-side-ab-feature-variations)
+- [Kittens or Puppers, the Critical Study We have all been waiting for](#kittens-or-puppers-the-critical-study-we-have-all-been-waiting-for)
+- [Get More Help](#get-more-help)
+- [Gotchas / FAQ / Ranting](#gotchas--faq--ranting)
+  - [General](#general)
+  - [studyUtils](#studyutils)
+  - [The lifecycle and deployment of the add-on once it gets released](#the-lifecycle-and-deployment-of-the-add-on-once-it-gets-released)
+  - [Legacy Add-ons](#legacy-add-ons-1)
+  - [s.t.m.o - sql.telemetry.mozilla.org](#stmo---sqltelemetrymozillaorg)
+    - [Where are my pings?](#where-are-my-pings)
+- [Glossary](#glossary)
+- [OTHER DOCS](#other-docs)
+- [`StudyUtils.jsm` api used in `bootstrap.js`](#studyutilsjsm-api-used-in-bootstrapjs)
+  - [Configuration](#configuration)
+  - [Lifecycle](#lifecycle)
+  - [Running](#running)
+  - [TODO](#todo)
+- [FIXES](#fixes)
+- [TODO](#todo-1)
+- [Template](#template)
+- [Getting QA of your add-ons](#getting-qa-of-your-add-ons)
+- [Links and References](#links-and-references)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## What is a Shield Study?
+
+**Shield Study Add-ons** do these actions:
+
+* implement variations (1+) of a feature
+* report common study and add-on lifecycle events to Telemetry
+* report study-specific data about how users react to and interact with a specific variations
+* respond coherently to add-on life-cycle events (`install`, `startup`, `disable`, `uninstall`).
+
+## tl;dr - Creating a new Shield Study add-on based on the template
+
+1. **One time**:
+
+   * Clone this directory
+
+   ```
+   git clone template
+   rm -rf {.git,docs}/
+   git init
+   ```
+
+   * install dependencies, including [`mozilla/shield-studies-addon-utils`][mozilla-ssau].
+
+```
+npm install
+```
+
+    * install **Firefox Nightly** for easier development
+
+2. Edit and examine files:
+
+   * `addon/bootstrap.js`
+   * `addon/Config.jsm`
+   * `package.json`
+   * `addon/lib/*`
+
+3. Build the legacy add-on xpi. Run **Nightly** with add-on
+
+   `npm run firefox`
```

Comment:
Do you want this as an inline `<code>` block (single backtick), or did you want to code-fence it and have it formatted like a `<pre>` block (three backticks, similar to the `npm install` on L73-75)?